archinstall/lib/installer.py

```diff
@@ -820,21 +820,24 @@ def add_systemd_bootloader(self, boot_partition :Partition, root_partition :Part
 		if not os.path.exists(f'{self.target}/boot/loader'):
 			os.makedirs(f'{self.target}/boot/loader')
 
+		loader_conf = f'{self.target}/boot/loader/loader.conf'
```

Comment:
Maybe we should use `pathlib.Path()` here?
```python
loader_conf = pathlib.Path(f'{self.target}/boot/loader/loader.conf')
```
And later use `with loader_conf.open("r") as loader` instead?
I think I prefer `if loader_conf.exists()` instead of a `try` & `catch` scenario.

Is there a reason we you'd prefer try and except?

Reply:
I'd second to use Path over strings

Reply:
> Is there a reason we you'd prefer try and except?

@Torxed , I'm glad you asked so that I can give an explanation for that change. There is a race condition with the current solution and the one you are proposing that the `try` ... `except` statement fixes.

Reply:
Oh yea that is True, never crossed my mind. I mean in most cases that would be an issue. For us that isn't really the case unless someone would go out of their way in a separate shell to destroy something. Which has been one of the major reasons for archinstall being single threaded *(with the slight exception of forks being used in certain places)* to avoid issues like those.

But I don't mind adhering to best practices. Maybe stick a short note there explaining why if possible (same in the other PR) because generally speaking `try-catch` statements are slower and less clean. Maybe that's just a fix idea I have and I need to change it. But a little note would go a long way to please me xD

Reply:
I was not aware that the `try` ... `except` statement was slower. My intention was bug free code.